### PR TITLE
provider/opsgenie: Descriptions for Teams

### PR DESCRIPTION
**Pull request by tombuildsstuff**
- [x] Updating the OpsGenie SDK to the latest and greatest
- [x] Adding the Description field to a Team
- [x] Acceptance & Import Tests
- [x] Documentation

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
Tests pass

```
TF_ACC=1 go test ./builtin/providers/opsgenie -v -run TestAccOpsGenie -timeout 120m
=== RUN   TestAccOpsGenieTeam_importBasic
--- PASS: TestAccOpsGenieTeam_importBasic (7.11s)
=== RUN   TestAccOpsGenieTeam_importWithUser
--- PASS: TestAccOpsGenieTeam_importWithUser (9.36s)
=== RUN   TestAccOpsGenieTeam_importWithUserComplete
--- PASS: TestAccOpsGenieTeam_importWithUserComplete (8.25s)
=== RUN   TestAccOpsGenieUser_importBasic
--- PASS: TestAccOpsGenieUser_importBasic (4.86s)
=== RUN   TestAccOpsGenieUser_importComplete
--- PASS: TestAccOpsGenieUser_importComplete (4.49s)
=== RUN   TestAccOpsGenieTeamName_validation
--- PASS: TestAccOpsGenieTeamName_validation (0.00s)
=== RUN   TestAccOpsGenieTeamRole_validation
--- PASS: TestAccOpsGenieTeamRole_validation (0.00s)
=== RUN   TestAccOpsGenieTeam_basic
--- PASS: TestAccOpsGenieTeam_basic (3.83s)
=== RUN   TestAccOpsGenieTeam_withUser
--- PASS: TestAccOpsGenieTeam_withUser (7.06s)
=== RUN   TestAccOpsGenieTeam_withUserComplete
--- PASS: TestAccOpsGenieTeam_withUserComplete (6.71s)
=== RUN   TestAccOpsGenieTeam_withMultipleUsers
--- PASS: TestAccOpsGenieTeam_withMultipleUsers (8.05s)
=== RUN   TestAccOpsGenieUserUsername_validation
--- PASS: TestAccOpsGenieUserUsername_validation (0.00s)
=== RUN   TestAccOpsGenieUserFullName_validation
--- PASS: TestAccOpsGenieUserFullName_validation (0.00s)
=== RUN   TestAccOpsGenieUserRole_validation
--- PASS: TestAccOpsGenieUserRole_validation (0.00s)
=== RUN   TestAccOpsGenieUser_basic
--- PASS: TestAccOpsGenieUser_basic (3.86s)
=== RUN   TestAccOpsGenieUser_complete
--- PASS: TestAccOpsGenieUser_complete (3.77s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/opsgenie	67.363s
```

File: builtin/providers/opsgenie/resource_opsgenie_team.go [hunk 1]
```diff
@@ -53,10 +58,12 @@ func resourceOpsGenieTeamCreate(d *schema.ResourceData, meta interface{}) error
...
 
 	name := d.Get("name").(string)
+	description := d.Get("description").(string)
 
 	createRequest := team.CreateTeamRequest{
-		Name:    name,
-		Members: expandOpsGenieTeamMembers(d),
+		Name:        name,
+		Description: description,
+		Members:     expandOpsGenieTeamMembers(d),
```
Comment (stack72) [inline builtin/providers/opsgenie/resource_opsgenie_team.go]: If we pass an empty string here is this ok?

File: builtin/providers/opsgenie/resource_opsgenie_team.go [hunk 2]
```diff
@@ -126,11 +134,13 @@ func resourceOpsGenieTeamRead(d *schema.ResourceData, meta interface{}) error {
...
 	name := d.Get("name").(string)
+	description := d.Get("description").(string)
 
 	updateRequest := team.UpdateTeamRequest{
-		Id:      d.Id(),
-		Name:    name,
-		Members: expandOpsGenieTeamMembers(d),
+		Id:          d.Id(),
+		Name:        name,
+		Description: description,
```
Comment (stack72) [inline builtin/providers/opsgenie/resource_opsgenie_team.go]: If we pass an empty string, is this ok?

**Comment by tombuildsstuff** [comment]
There's a test `TestAccOpsGenieTeam_basic ` which covers omiting the field - I've added one with an empty Description too just to ensure that's covered, but I believe that's already being done by HCL? (LMK if you want me to remove it)

To clarify - when I go from the following state:
```
resource "opsgenie_team" "test" {
  name = "example"
}
```

to the following state:
```
resource "opsgenie_team" "test" {
  name        = "example"
  description = ""
}
```

The plan is empty - presumably because the HCL parser ignores empty values. Is that sufficient, or is it worth adding some validation to the description field? In addition the field is marked as `omitempty` in the Go source

Tests pass:
```
$ TF_ACC=1 go test ./builtin/providers/opsgenie -v -run TestAccOpsGenie -timeout 120m
=== RUN   TestAccOpsGenieTeam_importBasic
--- PASS: TestAccOpsGenieTeam_importBasic (49.01s)
=== RUN   TestAccOpsGenieTeam_importWithEmptyDescription
--- PASS: TestAccOpsGenieTeam_importWithEmptyDescription (65.86s)
=== RUN   TestAccOpsGenieTeam_importWithUser
--- PASS: TestAccOpsGenieTeam_importWithUser (310.29s)
=== RUN   TestAccOpsGenieTeam_importWithUserComplete
--- PASS: TestAccOpsGenieTeam_importWithUserComplete (80.16s)
=== RUN   TestAccOpsGenieUser_importBasic
--- PASS: TestAccOpsGenieUser_importBasic (32.28s)
=== RUN   TestAccOpsGenieUser_importComplete
--- PASS: TestAccOpsGenieUser_importComplete (103.84s)
=== RUN   TestAccOpsGenieTeamName_validation
--- PASS: TestAccOpsGenieTeamName_validation (0.00s)
=== RUN   TestAccOpsGenieTeamRole_validation
--- PASS: TestAccOpsGenieTeamRole_validation (0.00s)
=== RUN   TestAccOpsGenieTeam_basic
--- PASS: TestAccOpsGenieTeam_basic (93.53s)
=== RUN   TestAccOpsGenieTeam_withEmptyDescription
--- PASS: TestAccOpsGenieTeam_withEmptyDescription (24.78s)
=== RUN   TestAccOpsGenieTeam_withUser
--- PASS: TestAccOpsGenieTeam_withUser (45.91s)
=== RUN   TestAccOpsGenieTeam_withUserComplete
--- PASS: TestAccOpsGenieTeam_withUserComplete (50.18s)
=== RUN   TestAccOpsGenieTeam_withMultipleUsers
--- PASS: TestAccOpsGenieTeam_withMultipleUsers (99.12s)
=== RUN   TestAccOpsGenieUserUsername_validation
--- PASS: TestAccOpsGenieUserUsername_validation (0.00s)
=== RUN   TestAccOpsGenieUserFullName_validation
--- PASS: TestAccOpsGenieUserFullName_validation (0.00s)
=== RUN   TestAccOpsGenieUserRole_validation
--- PASS: TestAccOpsGenieUserRole_validation (0.00s)
=== RUN   TestAccOpsGenieUser_basic
--- PASS: TestAccOpsGenieUser_basic (53.54s)
=== RUN   TestAccOpsGenieUser_complete
--- PASS: TestAccOpsGenieUser_complete (63.40s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/opsgenie	1071.909s
```

File: builtin/providers/opsgenie/resource_opsgenie_team.go (hunk 2 above)
Comment (tombuildsstuff) [inline builtin/providers/opsgenie/resource_opsgenie_team.go]: it's caught by the HCL (which picks it up as a null value)

File: builtin/providers/opsgenie/resource_opsgenie_team.go (hunk 1 above)
Comment (tombuildsstuff) [inline builtin/providers/opsgenie/resource_opsgenie_team.go]: yep - it's caught by the HCL (which picks it up as a null value) - there's also tests for omitting/empty/completed values

Comment (tombuildsstuff) [inline builtin/providers/opsgenie/resource_opsgenie_team.go]: yep - it's caught by the HCL (which picks it up as a null value) - there's also tests for omitting/empty/completed values

**Comment by stack72** [comment]
In that case - all LGTM :) Thanks @tombuildsstuff 

P.
